app/models/grda_warehouse/tasks/client_cleanup.rb

```diff
@@ -382,16 +383,31 @@ def choose_best_name dest_attr, source_clients
       dest_attr
     end
 
+    def choose_best_pronouns dest_attr, source_clients
+      non_blank = source_clients.select { |sc| sc[:pronouns].present? }
+      if non_blank.any?
+        # Copying the min_by, but I'd put more weight on the more recent
+        # record. Is min_by correct?
+        best = non_blank.min_by { |sc| sc[:DateCreated] }
+        dest_attr[:pronouns] = best[:pronouns]
+      else
+        dest_attr[:pronouns] = nil
+      end
+
+      dest_attr
+    end
+
     def choose_best_ssn dest_attr, source_clients
       # Get the best SSN (has value and quality is full or partial, oldest breaks the tie)
       non_blank_ssn = source_clients.select { |sc| sc[:SSN].present? }
-      if non_blank_ssn.any?
+      non_placeholder_ssn = non_blank_ssn.reject { |sc| sc[:SSN].match?(/x/i) }
+      if non_placeholder_ssn.any?
         best = nil
         [1, 2, 8, 9, 99].each do |dq_value|
-          best = non_blank_ssn.select { |sc| sc[:SSNDataQuality] == dq_value }.min_by { |sc| sc[:DateCreated] }
+          best = non_placeholder_ssn.select { |sc| sc[:SSNDataQuality] == dq_value }.min_by { |sc| sc[:DateCreated] }
           break if best.present?
         end
-        best = non_blank_ssn.min_by { |sc| sc[:DateCreated] } if best.blank?
+        best = non_placeholder_ssn.min_by { |sc| sc[:DateCreated] } if best.blank?
 
         dest_attr[:SSN] = best[:SSN]
```

Comment:
Eep sorry I didn't really provide any spec on the ticket for this one aside from "check ssns work with X's". I don't think this is what we want. In fact I think the logic that was already in place is correct. @eanders I think we were concerned that perhaps SSNs with Xs were being rejected here, but they weren't. And they shouldn't be, because they are valid.

We _could_ make the existing merge "smarter" by having is actually sort by the number of X's, and then by the creation date. So, something like `1111-22-0001, DataQuality: 1`  would be preferred over `XXX-XX-0001, DataQuality: 1`. That feels somewhat unnecessary though, the DQ field is supposed to specify the actual quality. What do you think @eanders ? This affects warehouse merging.

Reply:
I'll revert this and the spec.

Reply:
Thanks, sorry about that!